src/cms/forms.py

```diff
@@ -20,11 +23,32 @@ class PageForm(BleachableModelForm, JanewayTranslationModelForm):
 
     class Meta:
         model = models.Page
-        exclude = ('journal', 'is_markdown', 'content_type', 'object_id')
+        exclude = ('is_markdown', 'content_type', 'object_id')
 
     def __init__(self, *args, **kwargs):
+        self.request = kwargs.pop('request', None)
+        self.journal = self.request.journal if self.request else None
+        self.press = self.request.press if self.request else None
+
         super(PageForm, self).__init__(*args, **kwargs)
 
+        if self.instance:
+            journal_type = ContentType.objects.get(app_label="journal", model="journal")
+            if self.instance.content_type == journal_type:
+                self.journal = journal_type.get_object_for_this_type(pk=self.instance.object_id)
+            press_type = ContentType.objects.get(app_label="press", model="press")
+            if self.instance.content_type == press_type:
+                self.press = press_type.get_object_for_this_type(pk=self.instance.object_id)
```

Comment:
These queries may run eeven when `self.journal` and `self.press` are set at the start of `__init__`.

Reply:
Yeah that's by design. I can add a comment.